src/jit/lsrabuild.cpp

```diff
@@ -0,0 +1,3021 @@
+XX  specific methods for building.                                           XX
+XX                                                                           XX
+XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+*/
+
+#include "jitpch.h"
+#ifdef _MSC_VER
+#pragma hdrstop
+#endif
+
+#ifndef LEGACY_BACKEND // This file is ONLY used for the RyuJIT backend that uses the linear scan register allocator
+
+#include "lsra.h"
+
+//------------------------------------------------------------------------
+// LocationInfoListNodePool::LocationInfoListNodePool:
+//    Creates a pool of `LocationInfoListNode` values.
+//
+// Arguments:
+//    compiler    - The compiler context.
+//    preallocate - The number of nodes to preallocate.
+//
+LocationInfoListNodePool::LocationInfoListNodePool(Compiler* compiler, unsigned preallocate) : m_compiler(compiler)
+{
+    if (preallocate > 0)
+    {
+        size_t                preallocateSize = sizeof(LocationInfoListNode) * preallocate;
+        LocationInfoListNode* preallocatedNodes =
+            reinterpret_cast<LocationInfoListNode*>(compiler->compGetMem(preallocateSize, CMK_LSRA));
+
+        LocationInfoListNode* head = preallocatedNodes;
+        head->m_next               = nullptr;
+
+        for (unsigned i = 1; i < preallocate; i++)
+        {
+            LocationInfoListNode* node = &preallocatedNodes[i];
+            node->m_next               = head;
+            head                       = node;
+        }
+
+        m_freeList = head;
+    }
+}
+
+//------------------------------------------------------------------------
+// LocationInfoListNodePool::GetNode: Fetches an unused node from the
+//                                    pool.
+//
+// Arguments:
+//    l -    - The `LsraLocation` for the `LocationInfo` value.
+//    i      - The interval for the `LocationInfo` value.
+//    t      - The IR node for the `LocationInfo` value
+//    regIdx - The register index for the `LocationInfo` value.
+//
+// Returns:
+//    A pooled or newly-allocated `LocationInfoListNode`, depending on the
+//    contents of the pool.
+LocationInfoListNode* LocationInfoListNodePool::GetNode(LsraLocation l, Interval* i, GenTree* t, unsigned regIdx)
+{
+    LocationInfoListNode* head = m_freeList;
+    if (head == nullptr)
+    {
+        head = reinterpret_cast<LocationInfoListNode*>(m_compiler->compGetMem(sizeof(LocationInfoListNode)));
+    }
+    else
+    {
+        m_freeList = head->m_next;
+    }
+
+    head->loc      = l;
+    head->interval = i;
+    head->treeNode = t;
+    head->m_next   = nullptr;
+
+    return head;
+}
+
+//------------------------------------------------------------------------
+// LocationInfoListNodePool::ReturnNodes: Returns a list of nodes to the
+//                                        pool.
+//
+// Arguments:
```

Comment:
This method both take a list and mutates it.  The comment indicates that it is just returning a list.

Reply:
Again, these were all just moved as-is from lsra.cpp. But I will add method headers.